lib/Backend/GlobOpt.cpp

```diff
@@ -9611,8 +9621,10 @@ GlobOpt::OptConstFoldUnary(
             }
             else
             {
-                isInt = false;
-                fValue = -(FloatConstType)INT32_MIN;
+                // disable const fold for Math.abs(INT32_MIN) because it causes dst to be float type which
+                // could be different with previous type spec result in LoopPrePass. The latter does int
+                // type spec for Math.abs(INT32_MIN).
+                return false;
```

Comment:
You are right, the only advantage of continuing compilation is that if Math.abs(INT32_MIN) doesn't run in the jitted code, no bailout/rejit will be needed. I will switch to raise RejitException as simple workaround.

---
In reply to: [94718982](https://github.com/Microsoft/ChakraCore/pull/2267#discussion_r94718982) [](ancestors = 94718982)